executors.py

```diff
@@ -178,34 +195,42 @@ def execute(self) -> List[PermutationResult]:
             dimensions.remove(self.dimensions[-1])
 
         # iteration through positions like: n, 0, n-1, 1, n-2, 2, ...
-        for iteration, position in enumerate(chain(*zip(reversed(range(len(dimensions))), range(len(dimensions))))):
+        for iteration, target_position in enumerate(chain(*zip(reversed(range(len(dimensions))), range(len(dimensions))))):
             if self.fast and iteration == 2:
                 break
 
-            if position == mid:
+            if target_position == mid:
                 break
 
             results_per_dimension = list()
 
+            # for the current position - swap all the allowed dimensions and append all possible orders to the result set
             for dimension in dimension_pool:
                 original_position = resulting_order.index(dimension)
-
-                permutation = list(resulting_order)
-                permutation = swap(permutation, position, original_position)
-                permutation_result = self._evaluate_permutation(permutation)
-                permutation_results.append(permutation_result)
-                results_per_dimension.append(permutation_result)
-
-            if position > mid:
-                best_order = sorted(
-                    results_per_dimension,
-                    key=lambda r: r.ram_usage)[0]
-            else:
-                best_order = sorted(
-                    results_per_dimension,
-                    key=lambda r: r.median_query_time(self.view_name))[0]
-
-            resulting_order = list(best_order.dimension_order)
-            dimension_pool.remove(resulting_order[position])
+                dimension_target = resulting_order[target_position]
+
+                if (not self._check_swap_dim_with_str_to_last_position(dimension, target_position)
+                    and dimension_target in dimension_pool):
```

Comment:
while new line 208 checks for dimension in the allowed dims, this check is for the target dims, so we don't swap dims if we don't want to

Reply:
Understood